src/sagemaker_sklearn_extension/feature_extraction/sequences.py

```diff
@@ -393,30 +445,76 @@ def _hybrid_interpolation(x):
-        elif self.extraction_type == "all":
-            extraction_setting = ComprehensiveFCParameters()
-        else:
+        if self.extraction_type not in ["minimal", "efficient", "all"]:
             raise ValueError(
                 f"{self.extraction_type} is not a supported feature extraction option. Please choose one from "
                 f"the following options: [minimal, efficient, all]."
             )
+        min_settings = MinimalFCParameters()
         # Extract time series features from the dataframe
         # Replace any ``NaNs`` and ``infs`` in the extracted features with median/extreme values for that column
         tsfresh_features = extract_features(
             X_df_no_nans,
-            default_fc_parameters=extraction_setting,
+            default_fc_parameters=min_settings,
             column_id="id",
             column_sort="time",
             impute_function=impute,
+            n_jobs=0,
         )
+        self.min_settings_card = tsfresh_features.shape[1]
+        # Minimal features computed indepdently to ensure they go first in the output,
+        # this is needed to ensure their survival when filtering features
+        if self.extraction_type in ["efficient", "all"]:
+            if self.extraction_type == "efficient":
+                settings = EfficientFCParameters()
+            else:
+                settings = ComprehensiveFCParameters()
+            settings = {k: v for k, v in settings.items() if k not in min_settings}
+            tsfresh_features_extra = extract_features(
+                X_df_no_nans,
+                default_fc_parameters=settings,
+                column_id="id",
+                column_sort="time",
+                impute_function=impute,
+                n_jobs=0,
+            )
+            self.extra_settings_card = tsfresh_features_extra.shape[1]
+            tsfresh_features = pd.concat([tsfresh_features, tsfresh_features_extra], axis=1)
+
         # If X_df.dropna() dropped some observations entirely (i.e., due to all NaNs),
         # impute each tsfresh feature for those observations with the median of that tsfresh feature
         tsfresh_features_imputed = impute(tsfresh_features.reindex(pd.RangeIndex(X_df["id"].max() + 1)))
         return tsfresh_features_imputed, X_df
 
+    def _filter_features(self, tsfresh_features, mode="transform", transform_thresholds=None):
+        if self.expansion_threshold < self.min_settings_card:
+            raise ValueError(
+                f"Provided filter threshold(s) (= {self.expansion_threshold}) can not be smaller than "
+                f"number of features generated by minimal settings (= {self.min_settings_card})"
+            )
+        filter_order = np.arange(self.min_settings_card, tsfresh_features.shape[1])
+        random_state = np.random.get_state()
+        np.random.seed(self.feature_sampling_seed)
+        np.random.shuffle(filter_order)
+        np.random.set_state(random_state)
```

Comment:
Technically the 4 lines above work, but I think it will be easier to understand if you instead instantiate a RandomState object
https://numpy.org/doc/1.16/reference/generated/numpy.random.RandomState.html
as an element of this object. Then instead of messing with np.random.seed you can just use that state for .shuffle

Reply:
It's not very clear for me, how will the interface of generating pipes would work then. At the moment I only feed the number (seed) to the transformer constructor, should I feed an object instead?  This will require quite a lot of changes in PR tbh, as we never done it before. Let's keep it this way, we can clean it up later if we conclude on that RandomState way can be same clean.